### PR TITLE
fix: django 42 support added

### PR DESCRIPTION
**Pull request by zubairshakoorarbisoft**
Added django42 upgrade support, to have more context on it please refer to the following [issue](https://github.com/edx/upgrades/issues/147)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       matrix:
         os: [ubuntu-20.04]
         python-version: ['3.8']
-        toxenv: [quality, django32]
+        toxenv: [quality, django32, django42]
         exclude:
```
Comment (UsamaSadiq) [inline .github/workflows/ci.yml:18]: Let's remove this exclude section so quality checks also start to run.

Comment (zubairshakoorarbisoft) [reply .github/workflows/ci.yml:18]: ok

File: tox.ini [hunk 2]
```diff
@@ -8,6 +8,7 @@ setenv =
 deps = 
     django32: Django>=3.2,<4.0
     -r requirements/test.txt
+    django42: Django>=4.2,<4.3
 commands = 
     pytest --cov=django_sites_extensions
```
Comment (UsamaSadiq) [inline tox.ini:14]: Let's try including this line to fix coverage issue.
```suggestion
        coverage xml
```

**Comment by UsamaSadiq** [comment]
@zubairshakoorarbisoft any update on this PR?

**Comment by zubairshakoorarbisoft** [comment]
> @zubairshakoorarbisoft any update on this PR?

Not yet @UsamaSadiq , as we added `pycodestyle` in it so we decided to update its code first accordiing  `pycodestyle` standards that's why this issue is blocked due to that.

**Review by UsamaSadiq — APPROVED** [review]

**Pull request merged**